app/generate/route.ts

```diff
@@ -25,15 +25,26 @@ type CommitsResponse = GetResponseTypeFromEndpointMethod<
 
 export async function POST(req: Request) {
   const { prompt: url } = await req.json();
-  const changelog = await handleRequest(url);
-  return BraintrustAdapter.toAIStreamResponse(changelog);
+  const result = await handleRequest(url);
+  
+  if (result instanceof Response) {
+    return result;
+  }
```

Comment:
small style nit would be to throw an `Error` of a specific type instead, and handle that here. Also would use 400 instead of 404

E.g.

```
try {
   const result = await handleRequest(url);
   return BraintrustAdapter.toAIStreamResponse(result);
} catch (e) {
   if (e instanceof NoReleases) {
     new Response("This repository doesn't have any releases yet.", { status: 400 });
   }
   throw e;
}
```

Reply:
removed all the specific error handling and added support for repos with no releases instead